### DISCUSSION TITLE
Another way to close a window

### DISCUSSION THREAD

**Discussion by willser** [emilk/egui#1825] [Q&A]
Hello,
I am adding a window into my project.And I want to close it by a button.I noticed those code in this repository:
https://github.com/emilk/egui/blob/898f4804b7b998ffeb1ff9f457b935e1364d6827/egui/src/containers/window.rs#L69-L72

And it work fine if I click the `close button` which be created by egui self.But when I try to close this window by another way like this:
````rust

Window::new("Import from cURL")
    .resizable(true)
    .open(&mut self.show_curl_window)
    .collapsible(false)
    .show(ctx, |ui| {
        // something

        self.show_curl_window = true;
````
I get a compile error,because `show_curl_window` be borrowed tiwce.I am clear with it,so I try to use `refcell` or other way to solute it,but the `open` function only accept `&mut bool`.I have to add a field to fix it like those:

````rust
self.show_curl_window = self.show_curl_window && self.temp_show;
Window::new("Import from cURL")
    .resizable(true)
    .open(&mut self.show_curl_window)
    .collapsible(false)
    .show(ctx, |ui| {
        // something
        self.temp_show = false;

````
I do not think it'a great way.So do you have some good idea?

**Comment by breadfan** [comment]
Have you solved your problem?

**Reply by willser** [reply]
not yet😭

**Reply by breadfan** [reply]
Check my thread
